src/Server/ProxyV1Handler.cpp

```diff
@@ -0,0 +1,123 @@
+#include <Server/ProxyV1Handler.h>
+#include <Poco/Net/NetException.h>
+#include <Common/NetException.h>
+#include <Interpreters/Context.h>
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int NETWORK_ERROR;
+    extern const int SOCKET_TIMEOUT;
+    extern const int CANNOT_READ_FROM_SOCKET;
+    extern const int CANNOT_PARSE_INPUT_ASSERTION_FAILED;
+}
+
+void ProxyV1Handler::run()
```

Comment:
It's a new functionality, but I don't see any test for it. Is it possible to test it with our integration test?

Reply:
we don't have a PROXY client, there is some functional test for current implementation:
https://github.com/ClickHouse/ClickHouse/blob/03be5ffab76a5da47873c1cb2133b32637188788/tests/queries/0_stateless/01601_proxy_protocol.sh
but I couldn't implement it in integration environment - seems docker container doesn't have `nc` utility this implementation requires
but anyway I tested it locally and it seems to be working properly :)